# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/hyprpaper/hotfix-xdg-runtimedir.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- hyprpaper-0.7.0/src/ipc/Socket.cpp.vanilla	2024-06-25 00:35:32.400312222 +0200
+++ hyprpaper-0.7.0/src/ipc/Socket.cpp	2024-06-25 00:40:20.498321014 +0200
@@ -27,7 +27,7 @@
         const auto HISenv = getenv("HYPRLAND_INSTANCE_SIGNATURE");
         const std::string USERID = std::to_string(getpwuid(getuid())->pw_uid);
 
-        const auto USERDIR = "/run/user/" + USERID + "/hypr/";
+        const std::string USERDIR = std::string(getenv("XDG_RUNTIME_DIR")) + "/hypr/";
 
         std::string socketPath = HISenv ? USERDIR + std::string(HISenv) + "/.hyprpaper.sock" : USERDIR + ".hyprpaper.sock";
 
